File: src/main/scala/com/bigdata/day1/Person.scala
package com.bigdata.day1

/**
  * @author : wangshengyu 
  * @date : 2019/5/2 
  */
//private[bigdata] 只能在day2包下的访问
//private[bigdata] class Person {
class Person {
  //val修饰的变量只有只读属性，有getter()没有setter()
  //相当于Java中的final修饰的变量
  val id = "9527"

  // var修饰的变量既有getter()也有setter()
  var age = 18

  // 类私有字段，只能在类内部使用
  private var name = "唐伯虎"

  //对象私有字段，访问权限更加严格，Person的方法只能访问当前对象的字段
  private[this] val pet = "小强"

  def getPet: Unit ={
    println(pet)
  }
}

//Person类的伴生对象
object Person{
  def main(args: Array[String]): Unit = {
    val p = new Person
    println(p.name)
    p.getPet

  }
}

File: src/main/java/com/bigdata/redis/string/Person.java
package com.bigdata.redis.string;

import java.io.Serializable;

/**
 * @ Author: wangshengyu
 * @ Date: 2019/11/4 11:54
 */
public class Person implements Serializable {
    private static final long serialVersionUID = -9012113097419111583L;
    private String name;
    private int age;

    public Person(){}

    public Person(String name, int age){
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    @Override
    public String toString() {
        return "Product{" +
                "name='" + name + '\'' +
                ", age=" + age +
                '}';
    }
}
